packages/contracts/contracts/Unirep.sol

```diff
@@ -90,27 +98,36 @@ contract Unirep is IUnirep, VerifySignature {
      * e.g. to insert a non-zero data field in the state tree leaf
      **/
     function manualUserSignUp(
-        uint48 epoch,
-        uint256 identityCommitment,
-        uint256 stateTreeLeaf,
+        uint256[] calldata publicSignals,
```

Comment:
I think this function should not accept a proof. The idea with this is to allow attesters to define their own signup proof that is verified in their own contract. We can accept the `stateLeafIdentityHash`, and `data` then calculate the state tree leaf like `H(stateLeafIdentityHash, H(data))`.